src/api_manager/compute_platform.h

```diff
@@ -17,27 +17,15 @@
 
 namespace google {
 namespace api_manager {
-
 namespace compute_platform {
 
-enum ComputePlatform { UNKNOWN = 0, GAE_FLEX = 1, GCE = 2, GKE = 3 };
-
-inline const char *ToString(ComputePlatform p) {
-  switch (p) {
-    case GAE_FLEX:
-      return "GAE Flex";
-    case GCE:
-      return "GCE";
-    case GKE:
-      return "GKE";
-    case UNKNOWN:
-    default:
-      return "unknown";
-  }
-}
+// Compute platforms:
+extern const char kGaeFlex[];
```

Comment:
hmm.  this extern is c thing.  prefer to use c++

maybe convert namespace compute_platform into  a class and use static public members 

also you can use const std::string

you can take a look at this sample:  https://github.com/envoyproxy/envoy/blob/master/source/common/http/headers.h

Reply:
I took the first approach since we don't have singleton util :P